pkg/controller/registry/reconciler/configmap.go

```diff
@@ -513,5 +515,36 @@ func (c *ConfigMapRegistryReconciler) CheckRegistryServer(logger *logrus.Entry,
 	}
 
 	healthy = true
+	podsAreLive, e := detectAndDeleteDeadPods(logger, c.OpClient, pods, source.GetNamespace())
+	if !podsAreLive {
+		healthy = false
+		err = e
+	}
```

Comment:
Okay this seems like it's the best we can do. `detectAndDeleteDeadPods` attempts to delete and dead pods, and returns the errors if there were any, but returns true if it finds a healthy pod. 

In `CheckRegistryServer`, if we get `false` from `detectAndDeleteDeadPods`, then we set `healthy` to `false` and return `error` if there were any. 

If there were error deleting dead pods, but we found a healthy pod, we still return `healthy=true`. The next time the loop is run we'll attempt to delete the dead pods again.

Reply:
I think we need to propagate errors all the way up the stack regardless of the boolean value returned with it. If any error occurs, we are automatically considering things unhealthy.

Structure all of these like this:
```
boolVal, err := doThing()
if err != nil {
    return false, fmt.Errorf("error doing thing: %v", err)
}
return boolVal, nil
```